plugins/UsersManager/API.php

```diff
@@ -107,7 +107,16 @@ public function getUserPreference($userLogin, $preferenceName)
             return $optionValue;
         }
 
-        return $this->getDefaultUserPreference($preferenceName, $userLogin);
+        $defaultPreference = $this->getDefaultUserPreference($preferenceName, $userLogin);
+
+        // A `get*` method should not set anything. But in this case we do it to improve performance as it prevents
+        // us from loading all siteIds (which can be 50k or more) the next time this preference is requested.
+        // this method can be called quite often when generating links etc (to get defaultWebsiteId).
+        if ($defaultPreference !== false && $preferenceName === self::PREFERENCE_DEFAULT_REPORT) {
+            $this->setUserPreference($userLogin, $preferenceName, $defaultPreference);
+        }
```

Comment:
Possible with some work but currently do not have the time for it. How important is that? A task would add some overhead as in 99.99% of the runs there would be nothing to set but it wouldn't be any relevant overhead.